state/constraints.go

```diff
@@ -15,6 +15,15 @@ import (
 	"github.com/juju/juju/core/instance"
 )
 
+// Constraints represents the state of a constraints with an ID.
+type Constraints struct {
```

Comment:
This is adding a layer to a layer. There was previously only one use of `constraintsWithID`. Now that the ID requirement is expanded, we should just add DocID to `constraintsDoc`, embed that, and use `Constraints` in place of `constraintsWithID`.

Reply:
My initial PR added the ID to the constraintsDoc. It led to problems because the existing constraints CRUD code did not expect `bson: _id`. I can look into how to rewrite that.

Reply:
solved in 4f71dc335eec2d5c9b88f69215e5fa853a2cd55e and 2ac8c8b044036fa8c0abd0a71c3f297a9a975f5e